src/cpp/desktop/DesktopSessionLauncher.cpp

```diff
@@ -512,12 +512,40 @@ Error SessionLauncher::launchSession(const QStringList& argList,
    Error error = abendLogPath().removeIfExists();
    if (error)
       LOG_ERROR(error);
+   
+#ifdef __APPLE__
+   
+   // we need indirection through arch to handle arm64
+   if (sessionPath_.getFilename() == "rsession-arm64")
+   {
+      QStringList archArgList;
+      archArgList.append(QStringLiteral("-arm64"));
+      archArgList.append(argList);
+      
+      return parent_process_monitor::wrapFork(
+               boost::bind(launchProcess,
+                           "/usr/bin/arch",
+                           archArgList,
```

Comment:
This seems fine but I don't understand how the session path gets passed to `/usr/bin/arch` (which presumably launches the session?)

Reply:
This was my mistake -- thanks for catching that.